src/platform/plugins/shared/controls/public/control_group/init_controls_manager.ts

```diff
@@ -103,17 +93,17 @@ export function initControlsManager(
   }
 
   async function addNewPanel(
-    { panelType, initialState }: PanelPackage<{}, DefaultControlState>,
+    { panelType, serializedState }: PanelPackage<DefaultControlState>,
     index: number
   ) {
-    if ((initialState as DefaultDataControlState)?.dataViewId) {
-      lastUsedDataViewId$.next((initialState as DefaultDataControlState).dataViewId);
+    if ((serializedState?.rawState as DefaultDataControlState)?.dataViewId) {
+      lastUsedDataViewId$.next((serializedState!.rawState as DefaultDataControlState).dataViewId);
```

Comment:
nit: The `!` here is a bit of a code smell. Is there any way to avoid that? `lastUsedDataViewID` isn't required, it falls back to the default doesn't it?

Reply:
`!` is just needed because of the cast.